core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java

```diff
@@ -89,6 +94,22 @@ public BigQuerySqlDialect(SqlDialect.Context context) {
     return emulateNullDirectionWithIsNull(node, nullsFirst, desc);
   }
 
+  @Override public boolean castRequiredForStringOperand(RexCall node) {
+    if (super.castRequiredForStringOperand(node)) {
+      return true;
+    }
+    RexNode operand = node.getOperands().get(0);
+    RelDataType castType = node.type;
+    if (operand instanceof RexLiteral) {
+      if (SqlTypeFamily.NUMERIC.contains(castType)) {
+        return true;
+      }
+      return false;
+    } else {
+      return true;
+    }
```

Comment:
How about this?
```
return !(operand instanceof RexLiteral)
    || SqlTypeFamily.NUMERIC.contains(node.type);
```

Reply:
Looks good. Will do the change. Thank you for pointing it out.